tests/TestCase/Collection/Iterator/FilterIteratorTest.php

```diff
@@ -32,21 +33,16 @@ class FilterIteratorTest extends TestCase
     public function testFilter()
     {
         $items = new \ArrayIterator([1, 2, 3]);
-        $callable = $this->getMockBuilder(\stdClass::class)
-            ->addMethods(['__invoke'])
-            ->getMock();
-        $callable->expects($this->at(0))
-            ->method('__invoke')
-            ->with(1, 0, $items)
-            ->will($this->returnValue(false));
-        $callable->expects($this->at(1))
-            ->method('__invoke')
-            ->with(2, 1, $items)
-            ->will($this->returnValue(true));
-        $callable->expects($this->at(2))
-            ->method('__invoke')
-            ->with(3, 2, $items)
-            ->will($this->returnValue(false));
+        $callable = function ($key, $value, $itemArg) use ($items) {
```

Comment:
We could probably just simplify this to if $key === 2 return true otherwise return false. I don't know if we need to test internal values like this doing with the exceptions.

Reply:
I can shorten the methods up. I had the arguments reversed for key/value, and have added more assertions as the keys/values are part of the interface we provide.